tests/e2e/util.go

```diff
@@ -6200,3 +6200,148 @@ func getBlockDevSizeInBytes(f *framework.Framework, ns string, pod *v1.Pod, devi
 	output = strings.TrimSuffix(output, "\n")
 	return strconv.ParseInt(output, 10, 64)
 }
+
+// getVsanDPersistentVolumeClaimSpecWithStorageClass return the PersistentVolumeClaim
+// spec for vsanDirect datastore with specified storage class.
+func getVsanDPersistentVolumeClaimSpecWithStorageClass(namespace string, ds string,
```

Comment:
can you call `getPersistentVolumeClaimSpecWithStorageClass()` and then modify the returned spec for vsanD, this will help with maintenance

Reply:
I tried using getPersistentVolumeClaimSpecWithStorageClass() method and then modifying spec, but it isn't working, generate name value is getting appended with name value field and pod creation fails

Reply:
try resetting `generate name` field with the returned spec